src/core/lib/compression/compression_internal.cc

```diff
@@ -246,4 +250,34 @@ DefaultCompressionAlgorithmFromChannelArgs(const grpc_channel_args* args) {
   return absl::nullopt;
 }
 
+int DefaultGzipCompressionLevelFromChannelArgs(const grpc_channel_args* args) {
+  if (args == nullptr) return Z_DEFAULT_COMPRESSION;
```

Comment:
This entire function can be written using something like this:

```
return grpc_channel_arg_get_integer(
    GRPC_GZIP_COMPRESSION_LEVEL,
    {Z_DEFAULT_COMPRESSION, 0, INT_MAX});
```

Reply:
nice hit